### ISSUE TITLE
Answer  vs. Response: what's the difference?

### ISSUE THREAD

**Issue by gmmoura** [DNSOP/draft-ietf-dnsop-terminology-bis#70]
Thanks the authors for their work in making this draft.

I'm going through and I noticed that the term "answer" (45 occurrences) and "response" (47 occurrences)  are used interchangeably in the text.

I am OK with that, but they should be explicitly defined IMHO, to avoid confusion. 

E.g.: if I:
``$ dig A example.nl:``

```shell
;; ANSWER SECTION:
example.nl.             3600    IN      A       94.198.159.35

;; AUTHORITY SECTION:
example.nl.             3600    IN      NS      ex2.sidnlabs.nl.
example.nl.             3600    IN      NS      ex1.sidnlabs.nl.
```

  * Is the response *the entire DNS-layer packet(s)*, with all the  RRs in it? (including flags,etc)

  * Is an answer *each line* in both ANSWER and AUTHORITY section?

Maybe this is clear to many ppl, but it would be nice to write it down.

thanks

**Comment by crankycanuck** [comment]
I do not believe that we have used these terms "interchangeably", since 
they're not. Some responses are answers and all answers are responses.  
Some answers contain an Answer Section and some (negative answers, for 
instance) do not.

If that isn't clear, we perhaps ought to add text similar to that, though I 
thought we had some discussion of this. I didn't see it while looking on my 
phone just now, however.

A

--
Please excuse my clumbsy thums

**Comment by paulehoffman** [comment]
I have opened PR #71 to fix the places where I think we are really talking about responses, not answers.

**Comment by crankycanuck** [comment]
I think given how late this is in the draft, I'd like to suggest just adding a new bit of text rather than changing the terms, because the WG has been over this ground some.  I think it'll be easier to get WG agreement on some text describing the differences without trying to pick each case.  Dunno where to put it -- probably intro or something like that -- but here's a suggestion:


This document is sometimes unclear on the terms "response" and "answer".  The lack of clarity is partly because the DNS specifications themselves use the terms ambiguously.  In general, a response is any DNS message with the QR bit set (see [forward reference here]).  An answer is any response that attempts actually to answer the question posed in the corresponding query message.  Not all answers are authoritative (see [forward reference here]).  Some answers contain an Answer Section (see [xref?]), but not all do (such as a negative answer, [forward reference]).  Some other responses are not an answer at all, such as an error response.


Sorry not to submit this in the usual and proper githubby way, but I'm on vacation and jamming this in.

**Comment by paulehoffman** [comment]
Closed with #71, but might be reopened during IESG review if needed.
